README.md

```diff
@@ -360,12 +360,104 @@ As long as the string you pass to the constructor is a valid selector string the
 
 *Note: at this time there is no support for multiple HTMLElements as parameters. If you are passing an HTMLElement as the parameter rather than a selector string you cannot pass multiple controls. If you do wish to use multiple controls, pass the plugin a selector string instead.
 
+## Authoring Plugins
+
+Writing a plugin for Container is relatively simple but there are some differences based on whether it is going to be a built-in plugin, or external.
+
+Plugins for Container should take the form of an ES6 module, preferably with a named export. E.g.:
+```javascript
+export class ContainerPlugin() {...}
+```
+
+When developing a plugin for Container v2.x there are 3 main lifecycle hooks to keep in mind: `preload`, `init`, and `start`.
+
+`preload()`: Preload expects an async method or at least a returned promise. The method is passed the entire [PluginManager](https://github.com/SpringRoll/SpringRollContainer/blob/main/src/PluginManager.js) object. In general all plugins need to access is the [Bellhop client](https://github.com/SpringRoll/Bellhop), but the entire object is available should you require it.
+Note: if this method is missing or fails to resolve, Container will automatically discard your plugin and it will not be loaded in properly.
```

Comment:
```suggestion
*Note: if this method is missing or the returned promise is rejected, Container will automatically discard your plugin and it will not be loaded in properly.*
```